src/rgw/rgw_http_client.cc

```diff
@@ -435,10 +435,10 @@ static int do_curl_wait(CephContext *cct, CURLM *handle, int signal_fd)
     return -EIO;
   }
 
-  if (wait_fd.revents > 0) {
+  if (wait_fd.fd > 0) {
     uint32_t buf;
     ret = read(signal_fd, (void *)&buf, sizeof(buf));
-    if (ret < 0) {
+    if (ret < 0 && ret!=EAGAIN) {
```

Comment:
since we're in non-blocking mode, we could try to read more than 4 bytes as a potential optimization to avoid later wakeups. something like this?
```c++
     std::array<char, 256> buf;
     ret = read(signal_fd, buf.data(), buf.size());
```